riva_api/nlp.py

```diff
@@ -209,40 +233,58 @@ def analyze_intent(self, input_string: str, options: rnlp.AnalyzeIntentOptions)
 
         Args:
             input_string (:obj:`str`): a string which will be classified.
-            options (:obj:`riva_api.proto.riva_nlp_pb2.AnalyzeIntentOptions`): an intent options. You may find
-                 fields description
-                `here <https://docs.nvidia.com/deeplearning/riva/user-guide/docs/reference/protos/protos.html#riva-proto-riva-nlp-proto>`_.
-
+            options (:obj:`riva_api.proto.riva_nlp_pb2.AnalyzeIntentOptions`, `optional`,
+                 defaults to :obj:`riva_api.proto.riva_nlp_pb2.AnalyzeIntentOptions()`):
+                 an intent options. You may find fields description `here
+                <https://docs.nvidia.com/deeplearning/riva/user-guide/docs/reference/protos/protos.html#riva-proto-riva-nlp-proto>`_.
+                Defaults to an instance of :obj:`AnalyzeIntentOptions` created without parameters.
```

Comment:
I'm not familiar with the `AnalyzeIntent API`, but wanted to ask about the changed behavior based on the documentation. Is there a difference in behavior from the server when `options` is explicitly set to `None` vs an empty `AnalyzeIntentOptions`?

* If there is no difference, the documentation is fine as-is and you can resolve this comment!
* If there is a difference, I would suggest updating the documentation

```suggestion
                If options is `None`, an instance of :obj:`AnalyzeIntentOptions` without parameters is used.
```

* If there is a difference and `None` is a useful for an edge case,  perhaps update the signature with a default instead of `Optional` to allow for `None`.
```python
def analyze_intent(..., options = rnlp.AnalyzeIntentOptions(), ...)
```

Reply:
There is no difference. I didn't set a `rnlp.AnalyzeIntentOptions()` as default because it is mutable.